crates/compiler/load_internal/src/file.rs

```diff
@@ -2163,6 +2156,31 @@ fn worker_task<'a>(
                 // added. In that case, do nothing, and keep waiting
                 // until we receive a Shutdown message.
                 if let Some(task) = find_task(&worker, injector, stealers) {
+                    log!(
+                        ">>> {}",
+                        match &task {
+                            BuildTask::LoadModule { module_name, .. } => {
+                                format!("BuildTask::LoadModule({:?})", module_name)
+                            }
+                            BuildTask::Parse { header } => {
+                                format!("BuildTask::Parse({})", header.module_path.display())
+                            }
+                            BuildTask::CanonicalizeAndConstrain { parsed, .. } => format!(
+                                "BuildTask::CanonicalizeAndConstrain({})",
+                                parsed.module_path.display()
+                            ),
+                            BuildTask::Solve { module, .. } => {
+                                format!("BuildTask::Solve({:?})", module.module_id)
+                            }
+                            BuildTask::BuildPendingSpecializations { module_id, .. } => {
+                                format!("BuildTask::BuildPendingSpecializations({:?})", module_id)
+                            }
+                            BuildTask::MakeSpecializations { module_id, .. } => {
+                                format!("BuildTask::MakeSpecializations({:?})", module_id)
+                            }
+                        }
+                    );
```

Comment:
Oh this looks really good to have, I occasionally try to insert `dbg!` in places here but I never know what to print out that will be both useful and succinct.